### PR TITLE
Combobox improvements

### PR DESCRIPTION
**Pull request by Vlatombe**
* Support default value
* Support readonly mode
* Align markup with textbox
* Merge `ComboBoxSEC1525Test` into `ComboBoxTest` (follow-up of https://github.com/jenkinsci/jenkins/commit/00064be3661e687da7215523a3ed9a37a85f51a3)

### Proposed changelog entries

* Developer: improve the combobox component: support default value and readonly mode.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — COMMENTED** [review]
seems good, are you looking to use this somewhere? (i.e. to demonstrate it working?)

File: test/src/test/java/lib/form/ComboBoxTest.java
Comment (daniel-beck) [inline test/src/test/java/lib/form/ComboBoxTest.java:159]:
```suggestion
    public void testEnsureXssNotPossible() throws Exception {
```

Excess Snot? No thanks.

**Comment by Vlatombe** [comment]
@timja This is for a closed-source plugin I'm working on, however this can be easily demonstrated using the `ui-samples-plugin`. I can file a downstream for it if desired.

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
> @timja This is for a closed-source plugin I'm working on, however this can be easily demonstrated using the `ui-samples-plugin`. I can file a downstream for it if desired.

fine by me without, would be good to add a changelog entry though, it adds to the API for plugin developers

File: core/src/main/resources/lib/form/combobox.jelly
Comment (MRamonLeon) [inline core/src/main/resources/lib/form/combobox.jelly:47]:
```suggestion
      The default value of the combo box, in case both @value and 'instance[field]' are null.
```

**Review by MRamonLeon — APPROVED** [review]
minor suggestion

**Review by MRamonLeon — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by Vlatombe** [comment]
@timja Hang on, not sure why the tests are not passing anymore even though they seemed ok in [`e3f4b48` (#4939)](https://github.com/jenkinsci/jenkins/pull/4939/commits/e3f4b48798939c4a4d9473433368816a5e482301)

**Comment by timja** [comment]
weird

**Comment by Vlatombe** [comment]
@timja Got it. The `@TestExtension` had to match the test method name which had been renamed by the suggestion from @daniel-beck

**Pull request merged**
